fdbclient/S3BlobStore.actor.cpp

```diff
@@ -175,6 +176,13 @@ Reference<S3BlobStoreEndpoint> S3BlobStoreEndpoint::fromString(std::string const
 		if (prefix != LiteralStringRef("blobstore"))
 			throw format("Invalid blobstore URL prefix '%s'", prefix.toString().c_str());
 
+		Optional<std::string> proxyHost, proxyPort;
+		if (proxy.present()) {
+			StringRef p(proxy.get());
+			proxyHost = p.eat(":").toString();
+			proxyPort = p.eat().toString();
+		}
```

Comment:
Can you add a check here both host and port need to be present? Otherwise, throws an error.

And, this check should probably done earlier in `main` function, when parsing the command line.

Reply:
Added in main.

Reply:
Thanks. I think the check should also be added here as well----to catch potential misuse of APIs by other callers/binaries.

Reply:
Good point.